Add data source plugin dev docs

RiskeyL opened the pull request.

HanqingZ commented on en/getting-started/install-self-hosted/local-source-code.md:
````diff
@@ -58,6 +58,7 @@ pyenv global 3.11
     ```
     cd api
     ```
+2. MacOS only: install libmagic with `brew install libmagic`
````
Could you change the number to a symbol ">" since this is a notification.
```suggestion
> MacOS only: install libmagic with `brew install libmagic`
```

AllenWriter commented:
Updated: https://docs.dify.ai/getting-started/install-self-hosted/local-source-code

RiskeyL committed:
Documentation edits made through Mintlify web editor

Commit:
review zh docs

RiskeyL committed:
Documentation edits made through Mintlify web editor

RiskeyL committed:
Documentation edits made through Mintlify web editor

RiskeyL committed:
Documentation edits made through Mintlify web editor

RiskeyL committed:
Documentation edits made through Mintlify web editor

RiskeyL committed:
Documentation edits made through Mintlify web editor

RiskeyL committed:
Documentation edits made through Mintlify web editor

RiskeyL committed:
Documentation edits made through Mintlify web editor

RiskeyL committed:
Documentation edits made through Mintlify web editor

RiskeyL committed:
Documentation edits made through Mintlify web editor

RiskeyL committed:
Documentation edits made through Mintlify web editor

RiskeyL committed:
Documentation edits made through Mintlify web editor

alterxyz committed:
Docs tools: 1 succeeded, some failed

Rename operation failed: - Lang 'zh': File 'dify-docs/plugin-dev-zh/datasource-plugin.mdx' - Skipped (non-compliant):
Missing/empty critical frontmatter fields for renaming: dimensions.type.primary, dimensions.type.detail, dimensions.level, standard_title, language (expected for metadata, though not used in filename suffix anymore).
- Lang 'zh': File 'dify-docs/plugin-dev-zh/tool-oauth.mdx' - Skipped (non-compliant):
Missing/empty critical frontmatter fields for renaming: dimensions.type.primary, dimensions.type.detail, dimensions.level, standard_title, language (expected for metadata, though not used in filename suffix anymore).
- Lang 'en': File 'dify-docs/plugin-dev-en/datasource-plugin.mdx' - Skipped (non-compliant):
Missing/empty critical frontmatter fields for renaming: dimensions.type.primary, dimensions.type.detail, dimensions.level, standard_title, language (expected for metadata, though not used in filename suffix anymore).
- Lang 'en': File 'dify-docs/plugin-dev-en/tool-oauth.mdx' - Skipped (non-compliant):
Missing/empty critical frontmatter fields for renaming: dimensions.type.primary, dimensions.type.detail, dimensions.level, standard_title, language (expected for metadata, though not used in filename suffix anymore).
- Lang 'ja': File 'dify-docs/plugin-dev-ja/datasource-plugin.mdx' - Skipped (non-compliant):
Missing/empty critical frontmatter fields for renaming: dimensions.type.primary, dimensions.type.detail, dimensions.level, standard_title, language (expected for metadata, though not used in filename suffix anymore).
Docs JSON application failed: docs.json processed with 5 critical issue(s) reported.

Details of critical issues:
- Lang '简体中文': [SeriousWarning] File 'datasource-plugin.mdx' in 'plugin-dev-zh' is .mdx but does not match FILENAME_PATTERN. Skipping this file.
- Lang '简体中文': [SeriousWarning] File 'tool-oauth.mdx' in 'plugin-dev-zh' is .mdx but does not match FILENAME_PATTERN. Skipping this file.
- Lang 'English': [SeriousWarning] File 'datasource-plugin.mdx' in 'plugin-dev-en' is .mdx but does not match FILENAME_PATTERN. Skipping this file.
- Lang 'English': [SeriousWarning] File 'tool-oauth.mdx' in 'plugin-dev-en' is .mdx but does not match FILENAME_PATTERN. Skipping this file.
- Lang '日本語': [SeriousWarning] File 'datasource-plugin.mdx' in 'plugin-dev-ja' is .mdx but does not match FILENAME_PATTERN. Skipping this file.

RiskeyL committed:
Documentation edits made through Mintlify web editor

RiskeyL committed:
Documentation edits made through Mintlify web editor

RiskeyL committed:
Documentation edits made through Mintlify web editor

RiskeyL committed:
Documentation edits made through Mintlify web editor

RiskeyL committed:
Merge branch 'main' into data-source-plugin-dev-docs

alterxyz committed:
Docs tools: 1 succeeded, some failed

Rename operation failed: - Lang 'zh': File 'dify-docs/plugin-dev-zh/0211-getting-started-dify-tool.mdx' - Renaming error: Unexpected error: 'str' object has no attribute 'get'
- Lang 'zh': File 'dify-docs/plugin-dev-zh/datasource-plugin.mdx' - Skipped (non-compliant):
Missing/empty critical frontmatter fields for renaming: dimensions.type.primary, dimensions.type.detail, dimensions.level, standard_title, language (expected for metadata, though not used in filename suffix anymore).
- Lang 'zh': File 'dify-docs/plugin-dev-zh/tool-oauth.mdx' - Skipped (non-compliant):
Missing/empty critical frontmatter fields for renaming: dimensions.type.primary, dimensions.type.detail, dimensions.level, standard_title, language (expected for metadata, though not used in filename suffix anymore).
- Lang 'en': File 'dify-docs/plugin-dev-en/0211-getting-started-dify-tool.mdx' - Renaming error: Unexpected error: 'str' object has no attribute 'get'
- Lang 'en': File 'dify-docs/plugin-dev-en/datasource-plugin.mdx' - Skipped (non-compliant):
Missing/empty critical frontmatter fields for renaming: dimensions.type.primary, dimensions.type.detail, dimensions.level, standard_title, language (expected for metadata, though not used in filename suffix anymore).
- Lang 'en': File 'dify-docs/plugin-dev-en/tool-oauth.mdx' - Skipped (non-compliant):
Missing/empty critical frontmatter fields for renaming: dimensions.type.primary, dimensions.type.detail, dimensions.level, standard_title, language (expected for metadata, though not used in filename suffix anymore).
- Lang 'ja': File 'dify-docs/plugin-dev-ja/0211-getting-started-dify-tool.mdx' - Renaming error: Unexpected error: 'str' object has no attribute 'get'
- Lang 'ja': File 'dify-docs/plugin-dev-ja/datasource-plugin.mdx' - Skipped (non-compliant):
Missing/empty critical frontmatter fields for renaming: dimensions.type.primary, dimensions.type.detail, dimensions.level, standard_title, language (expected for metadata, though not used in filename suffix anymore).
Docs JSON application failed: docs.json processed with 5 critical issue(s) reported.

Details of critical issues:
- Lang '简体中文': [SeriousWarning] File 'datasource-plugin.mdx' in 'plugin-dev-zh' is .mdx but does not match FILENAME_PATTERN. Skipping this file.
- Lang '简体中文': [SeriousWarning] File 'tool-oauth.mdx' in 'plugin-dev-zh' is .mdx but does not match FILENAME_PATTERN. Skipping this file.
- Lang 'English': [SeriousWarning] File 'datasource-plugin.mdx' in 'plugin-dev-en' is .mdx but does not match FILENAME_PATTERN. Skipping this file.
- Lang 'English': [SeriousWarning] File 'tool-oauth.mdx' in 'plugin-dev-en' is .mdx but does not match FILENAME_PATTERN. Skipping this file.
- Lang '日本語': [SeriousWarning] File 'datasource-plugin.mdx' in 'plugin-dev-ja' is .mdx but does not match FILENAME_PATTERN. Skipping this file.

RiskeyL committed:
Documentation edits made through Mintlify web editor

alterxyz committed:
Docs tools: 1 succeeded, some failed

Rename operation failed: - Lang 'zh': File 'dify-docs/plugin-dev-zh/0211-getting-started-dify-tool.mdx' - Renaming error: Unexpected error: 'str' object has no attribute 'get'
- Lang 'zh': File 'dify-docs/plugin-dev-zh/datasource-plugin.mdx' - Skipped (non-compliant):
Missing/empty critical frontmatter fields for renaming: dimensions.type.primary, dimensions.type.detail, dimensions.level, standard_title, language (expected for metadata, though not used in filename suffix anymore).
- Lang 'zh': File 'dify-docs/plugin-dev-zh/tool-oauth.mdx' - Skipped (non-compliant):
Missing/empty critical frontmatter fields for renaming: dimensions.type.primary, dimensions.type.detail, dimensions.level, standard_title, language (expected for metadata, though not used in filename suffix anymore).
- Lang 'en': File 'dify-docs/plugin-dev-en/0211-getting-started-dify-tool.mdx' - Renaming error: Unexpected error: 'str' object has no attribute 'get'
- Lang 'en': File 'dify-docs/plugin-dev-en/datasource-plugin.mdx' - Skipped (non-compliant):
Missing/empty critical frontmatter fields for renaming: dimensions.type.primary, dimensions.type.detail, dimensions.level, standard_title, language (expected for metadata, though not used in filename suffix anymore).
- Lang 'en': File 'dify-docs/plugin-dev-en/tool-oauth.mdx' - Skipped (non-compliant):
Missing/empty critical frontmatter fields for renaming: dimensions.type.primary, dimensions.type.detail, dimensions.level, standard_title, language (expected for metadata, though not used in filename suffix anymore).
- Lang 'ja': File 'dify-docs/plugin-dev-ja/0211-getting-started-dify-tool.mdx' - Renaming error: Unexpected error: 'str' object has no attribute 'get'
- Lang 'ja': File 'dify-docs/plugin-dev-ja/datasource-plugin.mdx' - Skipped (non-compliant):
Missing/empty critical frontmatter fields for renaming: dimensions.type.primary, dimensions.type.detail, dimensions.level, standard_title, language (expected for metadata, though not used in filename suffix anymore).
Docs JSON application failed: docs.json processed with 5 critical issue(s) reported.

Details of critical issues:
- Lang '简体中文': [SeriousWarning] File 'datasource-plugin.mdx' in 'plugin-dev-zh' is .mdx but does not match FILENAME_PATTERN. Skipping this file.
- Lang '简体中文': [SeriousWarning] File 'tool-oauth.mdx' in 'plugin-dev-zh' is .mdx but does not match FILENAME_PATTERN. Skipping this file.
- Lang 'English': [SeriousWarning] File 'datasource-plugin.mdx' in 'plugin-dev-en' is .mdx but does not match FILENAME_PATTERN. Skipping this file.
- Lang 'English': [SeriousWarning] File 'tool-oauth.mdx' in 'plugin-dev-en' is .mdx but does not match FILENAME_PATTERN. Skipping this file.
- Lang '日本語': [SeriousWarning] File 'datasource-plugin.mdx' in 'plugin-dev-ja' is .mdx but does not match FILENAME_PATTERN. Skipping this file.
